### DISCUSSION TITLE
Including Templates in the Collections' Page

### DISCUSSION THREAD

**Discussion by zeekhuge** [outline/outline#4848] [Ideas]
So, each collection has a Description page attached to it which list all the Documents in it. I was thinking, it would be great the also show the list of Templates associated with that Collection.

**Comment by tommoor** [comment]
Yea, it would be nice to have a way to get to them from here. Are you thinking more about management? If you start a New Doc the option to apply a template should always be there.

**Reply by zeekhuge** [reply]
After giving it a little more thought, I think the ideal case would be as follows:
1. Each template should have following options
    1. Associated with `All` the collections
    2. Associate with selected collections (and multiple collections can be specified)
2. In the `Description` page attached to each Collection, the list of associated templates will show up.
3. When creating a new document - the template button should show associated Templates (showing other templates is fine as well).

The point here is, to create some "General" Templates (associated with `All`), that could be used throughout the collections.  
And then there are templates associated with collection - which basically tells the team - these are the only type of content allowed in a particular collection.

**Comment by zeekhuge** [comment]
I guess, along with the addition of `Template` section to the `Descriptions` page, another page with list of the `User's` documents would also be greatly helpful. 

This list may include the docs where the `User` was just an editor (and not the author) as well.

This will help with accountability & updating the doc, for each particular user . So the User will be able to review and update the docs they authored.

**Comment by zeekhuge** [comment]
Hey @tommoor : I have experience with the techstack & would love to contribute for this feature, with your help & guidance. Let me know the starting point.

**Reply by tommoor** [reply]
If you wanted to edit templates to not require a `collectionId` that seems totally reasonable, templates without that id would be considered workspace templates. Could be a good starting point?

**Reply by zeekhuge** [reply]
Hi @tommoor . Thank you for that starting point.

I just quickly glanced over the code and here's what I understood.
1. The server already handles `undefined` `collectionId` for a document.
2. In the collection-selection menu at the time of template-creation, we can just add a `Workspace Template` option, with `null` `collectionId`.

However, what I am a little confused with the route `collection/:id/new`. Because this route create a new doc. And Template is basically a doc (with `isTemplate` true in it). So, how do we handle the `:id` in the above route ?

Also, it may be very well possible that the solution is already in the code & I might have not looked into that, since I just did a rough scan. Let me know.

**Reply by tommoor** [reply]
It's a relatively recent change that documents no longer require a `collectionId` – templates were built prior to that. In terms of routing  there is also `/doc/new` for creating a doc without collection

**Reply by zeekhuge** [reply]
Hey @tommoor !
Was trying to run a local version for testing & decided to use magic-link for Auth. I started by (basically [followed this](https://docs.getoutline.com/s/hosting/doc/local-development-5hEhFRXow7))
1. Copying `.env.sample` to `.env`
2. Setting up the SMTP config in `.env`
3. `make up`
 
I tried various versions of `.env` (removed all auth-config except SMTP, etc) but the magic-link based auth never appeared. 
What's the right way to get started with local-development ? Specifically, how to setup Auth for local deployment ?

**Reply by tommoor** [reply]
I added a note here on provisioning a local user:

https://docs.getoutline.com/s/hosting/doc/local-development-5hEhFRXow7#h-authentication

**Reply by zeekhuge** [reply]
@tommoor Looks like that link is private. 
Somehow, I was under the impression that Email-magic link can be used as an authentication method for a fresh new deployment. Had to do a deep-dive into the code to find that there is an helper filter that filters out the Email `AuthProviders` (checks for some variable isEmailEnabled or something).

Anyway, I ultimately ended up configuring Azure (which I use in my production) with the localhost.
However, that ended up taking the time I was hoping to spend on Testing - Probably should make it explicit ?

**Reply by tommoor** [reply]
Replaced with the public link now.

**Comment by troplin** [comment]
Recently templates have been moved to settings in #5808. I was hoping for the complete opposite direction, i.e. make templates less special and allow templates anywhere in the "document tree" for better organization, like regular nested documents.

Maybe even a way to limit the scope of a template to a certain document or subtree. Many templates are not really "global" and only make sense in a very specific area.

**Reply by zeekhuge** [reply]
Totally agreed @troplin.
We even submitted a PR in the same direction - #6089, but it seems @tommoor has a different vision for templates.
